website/docs/src/content/cloud/storage/sql-database.mdx

```diff
@@ -0,0 +1,324 @@
+    }
+  },
+);
+```
+
+### Multi-statement scripts
+
+For complex database changes, you can execute multiple statements at once:
+
+```tsx
+const SetupUserSystem = gensx.Component("SetupUserSystem", async () => {
+  const db = await useDatabase("users");
+
+  // Execute a SQL script with multiple statements
+  await db.executeMultiple(`
+      CREATE TABLE IF NOT EXISTS users (
+        id TEXT PRIMARY KEY,
+        name TEXT NOT NULL,
+        email TEXT UNIQUE
+      );
+
+      CREATE TABLE IF NOT EXISTS user_preferences (
+        user_id TEXT PRIMARY KEY,
+        theme TEXT DEFAULT 'light',
+        notifications BOOLEAN DEFAULT 1,
+        FOREIGN KEY (user_id) REFERENCES users(id)
+      );
+
+      CREATE INDEX IF NOT EXISTS idx_users_email ON users(email);
+    `);
+
+  return "User system set up successfully";
+});
+```
+
+### Database schema migrations
+
+When you need to update your database schema, use migrations:
+
+```tsx
+const MigrateDatabase = gensx.Component(
+  "MigrateDatabase",
+  async ({ version }) => {
+    const db = await useDatabase("app_data");
+
+    if (version === "v2") {
+      // Run migrations with foreign key checks disabled
+      await db.migrate(`
+        ALTER TABLE products ADD COLUMN category TEXT;
+        CREATE TABLE product_categories (
+          id INTEGER PRIMARY KEY,
+          name TEXT NOT NULL,
+          description TEXT
+        );
+      `);
+
+      return "Database migrated to v2";
+    }
+
+    return "No migration needed";
+  },
+);
+```
+
+## Development vs. production
+
+GenSX SQL databases work identically in both local development and cloud environments:
+
+- **Local development**: Databases are stored as SQLite files in the `.gensx/databases` directory by default
+- **Cloud deployment**: Databases are automatically provisioned in the cloud
+
+No code changes are needed when moving from development to production.
+
+## Use cases
+
+### Data-backed agents
+
+Create agents that can query and update structured data:
+
+```tsx
+const DataAnalyst = gensx.Component("DataAnalyst", async ({ query }) => {
+  // Initialize the database with the baseball stats
+  await InitializeDatabase();
+
+  // Use the SQL Copilot to answer the question
+  return <SqlCopilot question={query} />;
+});
+```
+
+### User data storage
+
+Store user data and preferences in a structured format:
+
+```tsx
+const UserPreferences = gensx.Component(
+  "UserPreferences",
+  async ({ userId, action, data }) => {
+    const db = await useDatabase("user_data");
+
+    if (action === "get") {
+      const result = await db.execute(
+        "SELECT * FROM preferences WHERE user_id = ?",
+        [userId],
+      );
+      return result.rows.length > 0 ? result.rows[0] : null;
+    } else if (action === "set") {
+      await db.execute(
+        "INSERT OR REPLACE INTO preferences (user_id, settings) VALUES (?, ?)",
+        [userId, JSON.stringify(data)],
+      );
+      return { success: true };
+    }
```

Comment:
might not be worth the time to update, but this doesn't really feel any different than the blob storage example. Could easily change this so we have a few set user preference fields in the db and we set/query those so it shows differentation from just storing json

also, in it's current form, probably need to parse the json that comes back from the "get"

Reply:
I will leave this one for now. Agree it isn't the strongest example. There will be use cases for doing this via blobs and different ones for doing it via SQL thought I don't think we need to get into the nuances here. Just intended to be easy to scan.